src/ImageSharp.Formats.Jpeg/JpegDecoderCore.cs

```diff
@@ -187,7 +187,10 @@ public void Decode<TColor>(Image<TColor> image, Stream stream, bool configOnly)
             }
 
             // Process the remaining segments until the End Of Image marker.
-            while (true)
+            bool processBytes = true;
```

Comment:
You're not counting the mcu's anymore? This doesn't fix the issue now with the image in #18